cmd/jaeger/internal/sanitizer/utf8_test.go

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package sanitizer
+
+import (
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func invalidUTF8() string {
+	s, _ := hex.DecodeString("fefeffff")
+	return string(s)
+}
+
+func getBytesValueFromString(s string) pcommon.Value {
+	b := pcommon.NewValueBytes()
+	b.Bytes().Append([]byte(s)...)
+	return b
+}
+
+var utf8EncodingTests = []struct {
+	name          string
+	key           string
+	value         string
```

Comment:
It seems if your tests only allow string values you cannot test other attribute types.

Reply:
@yurishkuro I added some separate tests for non-string attributes